omnibus/partybus/lib/partybus/migration_api/v1.rb

```diff
@@ -97,19 +97,26 @@ def run_command(command, options={})
         runner.run_command(command, options)
       end
 
-      def run_sqitch(ec_target, osc_target="@1.0.4")
-        args =  " --db-user #{Partybus.config.postgres['db_superuser']}"
-        args << " --db-name opscode_chef"
-        args << " --db-host #{Partybus.config.postgres['vip']}"
-        args << " --db-port #{Partybus.config.postgres['port']}"
-        args << " --engine pg"
-
-        cmd = "cd /opt/opscode/embedded/service/opscode-erchef/schema"
-        cmd << " && cd baseline && sqitch #{args} deploy --to-target #{osc_target} --verify"
-        cmd << " && cd .. && sqitch #{args} deploy --to-target #{ec_target} --verify;"
-        run_command(cmd,
-                    :cwd => "/opt/opscode/embedded/service/opscode-erchef/schema",
-                    :env => {"PGPASSWORD" => Partybus.config.postgres['db_superuser_password']})
+      # credentials is a required hash containing
+      # EITHER: :service - name of key into 'secrets' that should contain
+      #                    sql_user and sql_password entries,
+      #    OR:  :username, :password for sql auth
+      def run_sqitch(target, path, database, credentials)
```

Comment:
I started down that path, but currently can't recall why I didn't stick with it.  In any case - that sounds good to me.